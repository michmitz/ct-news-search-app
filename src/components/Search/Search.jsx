import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ search, onSubmit, onChange }) => (
  <>
    <label htmlFor="search">Search</label>
    <input
      id="search"
      type="search"
      name="search"
      value={search}
      onChange={onChange}
    />
    <button onClick={onSubmit} data-testid="button">Search</button>
  </>
);

Search.propTypes = {
  search: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Search;
