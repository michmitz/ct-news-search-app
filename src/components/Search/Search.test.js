import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Search from './Search'

describe('Search component', () => {
  afterEach(() => cleanup());
  it('updates display on search', () => {
    const { asFragment } = render(<Search
      search=""
      onChange={() => {}}
      onSubmit={() => {}}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});