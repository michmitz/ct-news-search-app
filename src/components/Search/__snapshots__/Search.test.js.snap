// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component updates display on search 1`] = `
<DocumentFragment>
  <label
    for="search"
  >
    Search
  </label>
  <input
    id="search"
    name="search"
    type="search"
    value=""
  />
  <button
    data-testid="button"
  >
    Search
  </button>
</DocumentFragment>
`;
